Orange/data/table.py

```diff
@@ -204,11 +229,22 @@ def __new__(cls, *args, **kwargs):
         return cls.from_numpy(domain, *args, **kwargs)
 
     def __init__(self, *args, **kwargs):
-        # So subclasses can expect to call super without breakage; noop
-        pass
+        # TODO: use this instead of __new__
+        # all weights initialized to 1 (see the weight functions for details)
+        self[Table._WEIGHTS_COLUMN] = 1
+
+        self.name = kwargs.get("name") or self.name
```

Comment:
Table.name existed, but probably didn't work. I transferred it to remember its existence, we can throw it out later.